src/SubscribedStream.js

```diff
@@ -6,30 +6,60 @@ export default class SubscribedStream {
         this._client = client
         this.streamId = streamId
         this.subscriptions = {}
+        this.isPublisherPromises = {}
     }
 
-    getPublishers() {
+    async getPublishers() {
         if (!this.publishersPromise || (Date.now() - this.lastAccess) > PUBLISHERS_EXPIRATION_TIME) {
-            this.publishersPromise = this._client.getStreamPublishers(this.streamId)
+            this.publishersPromise = this._client.getStreamPublishers(this.streamId).then((publishers) => {
+                const map = {}
+                publishers.forEach((p) => {
+                    map[p] = true
+                })
+                return map
+            })
             this.lastAccess = Date.now()
         }
         return this.publishersPromise
     }
 
+    async isPublisher(publisherId) {
+        if (!this.isPublisherPromises[publisherId]) {
+            this.isPublisherPromises[publisherId] = this._client.isStreamPublisher(this.streamId, publisherId)
+        }
+        return this.isPublisherPromises[publisherId]
+    }
+
+    async isValidPublisher(publisherId) {
```

Comment:
What is the difference between `isPublisher` and `isValidPublisher` ?

Reply:
`isPublisher` calls the endpoint but stores the promise to avoid multiple calls to the endpoint when the function is called again before the promise is resolved.
`isValidPublisher` makes use of the 2 endpoints.

Reply:
If I understand correctly, the intention is that a user of this class would use `isValidPublisher` instead of `isPublisher`? If this is true, would it make sense to rename `isPublisher` to `_isPublisher` to communicate that it is internal in nature. Or is there a case where a user of this class would want to use `isPublisher` instead of `isValidPublisher`?

Reply:
It is internal so yes I can prepend its name with `_`